app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -101,6 +102,8 @@ def check_password
     return if !@assessment.password_protected? || can?(:manage, @assessment)
 
     unless authentication_service.authenticated?
+      log_service.log_submission_access(request)
```

Comment:
This is actually a failed attempt right ?  how can users differentiate with success attempt? 

Also we should further log when user inputs password correctly. that time the session id also changes.